src/account/UpgradeableModularAccount.sol

```diff
@@ -623,10 +618,64 @@ contract UpgradeableModularAccount is
     // solhint-disable-next-line no-empty-blocks
     function _authorizeUpgrade(address newImplementation) internal override {}
 
-    function _checkIfValidationApplies(bytes4 selector, FunctionReference validationFunction, bool isDefault)
-        internal
-        view
-    {
+    function _checkIfValidationAppliesCallData(
+        bytes calldata callData,
+        FunctionReference validationFunction,
+        bool isDefault
+    ) internal view {
+        bytes4 outerSelector = bytes4(callData[:4]);
+        if (outerSelector == this.executeUserOp.selector) {
+            // If the selector is executeUserOp, pull the actual selector from the following data,
+            // and trim the calldata to ensure the self-call decoding is still accurate.
+            callData = callData[4:];
+            outerSelector = bytes4(callData[:4]);
```

Comment:
@adamegyed Is this line redundant (same as line 626)? Or do you want to pull the actual selector from `bytes4(userOp.callData[4:8])`?